src/convention/coerce.jl

```diff
@@ -168,3 +168,36 @@ function _check_tight(v::CArr, T, tight)
     end
     return v
 end
+
+
+## Images
+F = Union{Float32, FixedPointNumbers.Normed{UInt8,8}}
+_2d{C} =AbstractArray{C} where C<:Union{ColorTypes.AbstractRGB, ColorTypes.Gray}
+
+# Single Image
+function coerce(y::Arr{<:F, 2}, T2::Type{_2d{ColorTypes.Gray}})
+    return ColorTypes.Gray.(y)
+end
+
```

Comment:
The second argument of `coerce` is alwarys a scitype. In this case the calls will look something like `coerce(matrix, GrayImage)`or `coerce(matrix, ColorImage)`, so here you want:

```julia
function coerce(y::Arr{<:F, 2}, T2::Gray)
```